flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -140,14 +144,15 @@ else if (barrierId > currentCheckpointId) {
 
 				// abort the current checkpoint
 				releaseBlocksAndResetBarriers();
-				checkpointAborted = true;
+				notifyCheckpointCompletedOrCanceled(currentCheckpointId);
 
 				// begin a new checkpoint
 				beginNewAlignment(barrierId, channelIndex, receivedBarrier.getTimestamp());
 			}
 			else {
```

Comment:
we can reduce this path for simple by https://github.com/apache/flink/pull/11351/files#r397598435